sql/2020/06_Third_Parties/zero_third_party_websites.sql

```diff
@@ -0,0 +1,42 @@
+#standardSQL
+# Websites with no third party requests
+
+# Provides incorrect information in some cases, e.g. pageid = 140607555 
+
+WITH requests AS (
+  SELECT
+    pageid AS page,
+    req_host AS host
+  FROM
+    `httparchive.summary_requests.2020_08_01_mobile`
+),
+pages AS (
+  SELECT
+    pageid,
+    url
+  FROM
+    `httparchive.summary_pages.2020_08_01_mobile`
+),
+base AS (
+  SELECT
+    LOGICAL_AND(IF(NET.HOST(host) = NET.HOST(url), TRUE, FALSE)) zero_third_party,
+    url
+  FROM
+    requests
+  INNER JOIN
+    pages
+  ON
+    requests.page = pages.pageid
+  GROUP BY
+    url
+)
+
+SELECT
+  url
+FROM
+  base
+WHERE
+  zero_third_party = TRUE
+  AND REGEXP_CONTAINS(url, r"^https:\/\/www\.[a-z]+\.com\/$")
```

Comment:
Could you elaborate on why this is needed?

Reply:
Wanted to narrow down the search for zero-third-party to most probably valid and operating websites. Will remove it as soon as we are able to filter out crawling issues (if they are caused at a crawl stage).

Reply:
Did another approach, removed domain regex.